src/Microsoft.AspNet.Owin/OwinExtensions.cs

```diff
@@ -67,21 +68,32 @@ public static IApplicationBuilder UseOwin(this IApplicationBuilder builder, Acti
 
         public static IApplicationBuilder UseBuilder(this AddMiddleware app)
         {
+            return app.UseBuilder(configureServices: null);
+        }
+
+        public static IApplicationBuilder UseBuilder(this AddMiddleware app, Action<IServiceCollection> configureServices)
```

Comment:
Shouldn't this return a `IServiceProvider` i.e. `Func<IServiceProvider>`? How would you configure a different IoC container?